pkg/agents/team5/agent1/agent.go

```diff
@@ -0,0 +1,174 @@
+package team5
+
+import (
+	"math"
+
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/SOMAS2021/SOMAS2021/pkg/utils/globalTypes/food"
+)
+
+// TODO: Requires message passing
+// type Memory struct {
+// 	trust             float64 // scale of -5 to 5, with -5 being least trustworthy and 5 being most trustworthy, 0 is neutral
+// 	favour            float64 // e.g. generosity; scale of -5 to 5, with -5 being least favoured and 5 being most favoured, 0 is neutral
+// 	daysSinceLastSeen int     // days since last interaction
+// }
+
+type CustomAgent5 struct {
+	*infra.Base
+	selfishness       int
+	lastMeal          food.FoodType
+	daysSinceLastMeal int
+	// TODO: Change this to an enum
+	currentAim   int
+	satisfaction int
+	daysAlive    int
+	// TODO: Check difference between this and HasEaten()
+	// If true, then agent will attempt to eat
+	attemptToEat bool
+	// TODO: Requires message passing
+	// Social network of other agents
+	// memory map[string]Memory
+}
+
+func New(baseAgent *infra.Base) (infra.Agent, error) {
+	return &CustomAgent5{
+		Base:              baseAgent,
+		selfishness:       3,    // of 0 to 3, with 3 being completely selfish, 0 being completely selfless
+		lastMeal:          0,    //Stores value of the last amount of food taken
+		daysSinceLastMeal: 0,    //Count of how many days since last eating
+		currentAim:        0,    //Scale of 0 to 2, 0 being willing to lose health, 1 being maintaining health, 2 being gaining health
+		satisfaction:      0,    //Scale of -3 to 3, with 3 being satisfied and unsatisfied
+		daysAlive:         0,    //Count how many days agent has been alive
+		attemptToEat:      true, //Variable needed to check if we have already attempted to eat on a day
+		// TODO: Requires message passing
+		// memory:            map[string]Memory{}, // Memory of other agents, key is agent id
+	}, nil
+}
+
+// TODO: Requires message passing
+// func (a *CustomAgent5) newMemory(id string) {
+// 	a.memory[id] = Memory{
+// 		trust:             0,
+// 		favour:            0,
+// 		daysSinceLastSeen: 0,
+// 	}
+// }
+
+func (a *CustomAgent5) updateAim() {
+	switch {
+	case a.selfishness >= 3:
+		// If fully selfish always try to gain health
+		a.currentAim = 2
+	case a.HP() > 80 && a.selfishness == 2:
```

Comment:
Maybe worth packaging these thresholds into a variable + function of maxHP for debugging / posterity

This is your decision though, since doesn't affect anything else :)

Reply:
Will merge for now but you're right, thanks